src/main/java/org/seqdoop/hadoop_bam/BAMRecordReader.java

```diff
@@ -207,10 +213,36 @@ public static long getKey0(int refIdx, int alignmentStart0) {
 				}
 			}
 
+			if (!overlaps(r, intervals)) {
```

Comment:
We should probably keep an eye on this line whether it becomes a bottleneck when we have many intervals to filter from and then possibly switch data structures for the intervals. But let's leave it as is for now.